assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -2224,6 +2225,27 @@ private String build_student_view_submission_confirmation_context(VelocityPortle
         context.put("value_ASSIGNMENT_INPUT_ADD_TIME_SPENT", state.getAttribute(ResourceProperties.ASSIGNMENT_INPUT_ADD_TIME_SPENT));
         context.put("isTimesheet", assignmentService.isTimeSheetEnabled((String) state.getAttribute(STATE_CONTEXT_STRING)));
 
+        try {
+            // Get site ID
+            String siteId = toolManager.getCurrentPlacement().getContext();
+            Optional<Site> site = Optional.of(siteService.getSite(siteId));
+            Site currentSite = site.get();
+            // Assignments Tool Configuration
+            ToolConfiguration toolConfig = currentSite.getToolForCommonId(TOOL_ID);
+                    
+            Properties roleConfig = toolConfig.getConfig();
+            // Get visibility value of assignments' tool
+            String visibility = roleConfig.getProperty(ToolManager.PORTAL_VISIBLE);
+            
+            if(!"false".equals(visibility)) {
```

Comment:
context.put("visible", StringUtils.equalsIgnoreCase(visibility, BOOLEAN.TRUE.toString()));

Reply:
Add a comment:

// Checks if the assignments tool is visible from the LHS Menu.